### PR TITLE
chore: modernize Ruff config

### PR DESCRIPTION
**Pull request by henryiii**
Modernizing the Ruff config a bit. More things are in a .lint namespace now.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -39,7 +35,8 @@ repos:
     rev: v0.1.3
     hooks:
       - id: ruff
-        args: [--fix, --exit-non-zero-on-fix, --unsafe-fixes]
+        args: [--fix, --exit-non-zero-on-fix, --unsafe-fixes, --show-fixes]
+      - id: ruff-format
```
Comment (gaborbernat) [inline .pre-commit-config.yaml:39]: Shouldn't the format be before the lint?

File: pyproject.toml [hunk 2]
```diff
@@ -61,18 +61,20 @@ build.targets.sdist.include = ["/src", "/tests","tox.ini"]
...
 [tool.ruff]
-select = ["ALL"]
 line-length = 120
-target-version = "py38"
-isort = {known-first-party = ["pyproject_fmt"], required-imports = ["from __future__ import annotations"]}
+src = ["src", ""]
+
+[tool.ruff.lint]
+select = ["ALL"]
+isort.required-imports = ["from __future__ import annotations"]
```
Comment (gaborbernat) [inline pyproject.toml:69]: I strongly prefer to inline things rather than define new tables.

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (henryiii) [reply .pre-commit-config.yaml:39]: The formatter is guaranteed to not add new violations (if not, it's a bug and can be reported), while the linter is not guaranteed to not mess up the format. This is the official recommendation for now, eventually there might be a more elegant solution.

File: pyproject.toml (hunk 2 above)
Comment (henryiii) [reply pyproject.toml:69]: Okay.

Comment (henryiii) [reply pyproject.toml]: I wasn't sure, per-file-ignores is a separate table already.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (gaborbernat) [reply .pre-commit-config.yaml:39]: But if I run the linter, and the format; wouldn't it mean my linter reports will get reported with bad lines?

Comment (henryiii) [reply .pre-commit-config.yaml:39]: That’s why there will be a better solution in the future. I assume it likely will have to be a way to run both at once. (But that’s a guess)

Comment (henryiii) [reply .pre-commit-config.yaml:39]: https://x.com/henryschreiner3/status/1717232907235086710?s=46&t=LsA5Nji4tS3d8uQSfmTybw

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**